web/src/lib/components/shared-components/search-bar/search-history-box.svelte

```diff
@@ -2,51 +2,130 @@
+  $: isSearchSuggestions = filteredSearchTerms.length > 0;
+  $: showClearAll = searchQuery === '';
+  $: suggestionCount = showClearAll ? filteredSearchTerms.length + 1 : filteredSearchTerms.length;
+
+  let selectedIndex: number | undefined = undefined;
+  let element: HTMLDivElement;
+
+  export function moveSelection(increment: 1 | -1) {
+    if (!isSearchSuggestions) {
+      return;
+    } else if (selectedIndex === undefined) {
+      selectedIndex = increment === 1 ? 0 : suggestionCount - 1;
+    } else if (selectedIndex + increment < 0 || selectedIndex + increment >= suggestionCount) {
+      clearSelection();
+    } else {
+      selectedIndex = (selectedIndex + increment + suggestionCount) % suggestionCount;
+    }
+    onActiveSelectionChange(getId(selectedIndex));
+  }
+
+  export function clearSelection() {
+    selectedIndex = undefined;
+    onActiveSelectionChange(undefined);
+  }
+
+  export function selectActiveOption() {
+    if (selectedIndex === undefined) {
+      return;
+    }
+    const selectedElement = element.querySelector(`#${getId(selectedIndex)}`) as HTMLElement;
+    selectedElement?.click();
+  }
+
+  const handleClearAll = () => {
+    clearSelection();
+    onClearAllSearchTerms();
+  };
 
-  {#each $savedSearchTerms as savedSearchTerm, i (i)}
+  const handleClearSingle = (searchTerm: string) => {
+    clearSelection();
+    onClearSearchTerm(searchTerm);
+  };
+
+  const handleSelect = (searchTerm: string) => {
+    clearSelection();
+    onSelectSearchTerm(searchTerm);
+  };
+
+  const getId = (index: number | undefined) => {
+    if (index === undefined) {
+      return undefined;
+    }
+    return `${id}-${index}`;
+  };
+</script>
+
+<div role="listbox" {id} aria-label={$t('recent_searches')} bind:this={element}>
+  {#if isOpen && isSearchSuggestions}
     <div
-      class="flex w-full items-center justify-between text-sm text-black hover:bg-gray-100 dark:text-gray-300 dark:hover:bg-gray-500/10"
+      transition:fly={{ y: 25, duration: 150 }}
+      class="absolute w-full rounded-b-3xl border-2 border-t-0 border-gray-200 bg-white pb-5 shadow-2xl transition-all dark:border-gray-600 dark:bg-immich-dark-gray dark:text-gray-300"
     >
-      <div class="relative w-full items-center">
-        <button
-          type="button"
-          class="relative flex w-full cursor-pointer gap-3 py-3 pl-5"
-          on:click={() => dispatch('selectSearchTerm', savedSearchTerm)}
-        >
-          <Icon path={mdiMagnify} size="1.5em" />
-          {savedSearchTerm}
-        </button>
-        <div class="absolute right-5 top-0 items-center justify-center py-3">
-          <button type="button" on:click={() => dispatch('clearSearchTerm', savedSearchTerm)}
-            ><Icon path={mdiClose} size="18" /></button
+      <div class="flex items-center justify-between px-5 pt-5 text-xs">
+        <p class="py-2" aria-hidden={true}>{$t('recent_searches').toUpperCase()}</p>
+        {#if showClearAll}
+          <button
```

Comment:
I wanted to find a way to give keyboard users the ability to clear recent search options, while keeping the UI the same and making the keyboard interactions simple to allow for the search functionality to change down the road. For example, allowing searching on a user's albums or immich pages (ex. Favorites, Explore, Archive, etc).

This solution is generally inspired by how LinkedIn implemented their search combobox. In my opinion, I like that it makes the clearing functionality easily available without adding additional tab stops, and it's in the flow of a normal dropdown menu that users are already familiar with using.

A downside of this solution is that the button to clear a single search option is currently not keyboard accessible. An idea for this is to make the `tabindex` of the remove single button dynamic - if that option is `aria-selected`, set it to 0, else set it to -1. Bing search works similar to this.

I'm open to feedback or ideas for different ways to implement this. Adding the "Clear all" button as the first option certainly isn't the only solution.

Reply:
> This solution is generally inspired by how LinkedIn implemented their search combobox. In my opinion, I like that it makes the clearing functionality easily available without adding additional tab stops, and it's in the flow of a normal dropdown menu that users are already familiar with using.

I agree with that, but the `Clear all` button being the first listbox item is confusing because it's announced as "recent searches list - clear all recent searches one of three" I think it makes more sense how LinkedIn does it by putting the button outside the listbox, where it's simply announced as "clear search history button".

Reply:
> I think it makes more sense how LinkedIn does it by putting the button outside the listbox, where it's simply announced as "clear search history button".

Yeah, I agree that announcing the button without the count is easier to understand. Thanks for your feedback, I will fix this!

Reply:
FYI @michelheusschen and @alextran1502, I will be working on a follow-up PR for this next week to resolve this feedback, as I didn't get the fix in before it was merged. 

I am not 100% happy with how the screen reader is announcing it yet, and I have some code in progress.